jenkins/image_building/dib_elements/ubuntu-ci/post-install.d/02_configure

```diff
@@ -0,0 +1,161 @@
+for i in "${!SSH_VALUES[@]}"; do
+    name="${i}"
+    value="${SSH_VALUES[$i]}"
+    set_value "${name}" "${value}" /etc/ssh/sshd_config " "
+done
+
+# Set the permissions on the sshd_config file so that only root users can change its contents
+sudo chown root:root /etc/ssh/sshd_config
+sudo chmod 600 /etc/ssh/sshd_config
+
+# Loop through networking table
+for i in "${!NETWORK_PARAMETERS[@]}"; do
+  name="${i}"
+  value="${NETWORK_PARAMETERS[$i]}"
+  set_value "${name}" "${value}" /etc/sysctl.conf "="
+done
+
+# Remove legacy services
+sudo apt-get --purge remove telnet -y
+sudo apt-get autoremove -y
+
+# We do not use passwords on the machines
+
+# Disable the system accounts for non-root users
+
+# shellcheck disable=SC2013
+for user in $(awk -F: '($3 < 500) {print $1 }' /etc/passwd); do
+  if [[ "${user}" != "root" ]]; then
+    sudo /usr/sbin/usermod -L "${user}"
+if [[ "${user}" != "sync" ]] && [[ "${user}" != "shutdown" ]] && [[ "${user}" != "halt" ]]; then
+      sudo /usr/sbin/usermod -s /sbin/nologin "${user}"
+    fi
+  fi
+done
+
+# Set User/Group Owner and Permission on “/etc/anacrontab”, “/etc/crontab” and “/etc/cron
+sudo chown root:root /etc/crontab /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d
+sudo chmod og-rwx /etc/crontab /etc/cron.hourly /etc/cron.daily /etc/cron.weekly /etc/cron.monthly /etc/cron.d
+
+# Set the right and permissions on root crontab
+sudo chown root:root /var/spool/cron/crontabs
+sudo chmod og-rwx /var/spool/cron/crontabs
+
+# Set User/Group Owner and Permission on “passwd” and “group”  files
+sudo chmod 644 /etc/passwd /etc/group
+sudo chown root:root /etc/passwd /etc/group
+
+
+#Set User/Group Owner and Permission on the “shadow” and “gshadow” files
+sudo chmod 600 /etc/shadow /etc/gshadow
+sudo chown root:root /etc/shadow /etc/gshadow
+
+# Restrict Core Dumps
+echo '* hard core 0' | sudo tee -a /etc/security/limits.conf > /dev/null
+
+# MONiTORING. Collect monitoring data with atop and sar
+# https://aws.amazon.com/premiumsupport/knowledge-center/ec2-linux-configure-monitoring-tools/
+
+## Install monitoring tools
+sudo apt-get install atop sysstat -y
+
+## Collect all metrics every minute
+sudo sed -i 's/^LOGINTERVAL=600.*/LOGINTERVAL=60/' /usr/share/atop/atop.daily
+sudo sed -i -e 's|5-55/10|*/1|' -e 's|every 10 minutes|every 1 minute|' -e 's|debian-sa1|debian-sa1 -S XALL|g' /etc/cron.d/sysstat
+sudo bash -c "echo 'SA1_OPTIONS=\"-S XALL\"' >> /etc/default/sysstat"
+
+## Reduce metrics retention to 3 days
+sudo sed -i 's/^LOGGENERATIONS=.*/LOGGENERATIONS=3/' /usr/share/atop/atop.daily
+sudo sed -i 's/^HISTORY=.*/HISTORY=3/' /etc/default/sysstat
+
+## Enable services
+sudo sed -i 's|ENABLED="false"|ENABLED="true"|' /etc/default/sysstat
+sudo systemctl enable atop.service cron.service sysstat.service
+
+# RESET CLOUD INIT
+# Following will remove any cloud init's previous run
+# data and force cloud-init to again on next boot.
+
+sudo rm -rf /var/lib/cloud/*
```

Comment:
Same here, I don't think we need this.

Reply:
I will remove the prepull script. As I said it is not fully inlemented and causing a lot of confusion.